internal/addrs/move_endpoint.go

```diff
@@ -299,3 +299,18 @@ func (e *MoveEndpoint) internalAddrType() TargetableAddrType {
 		panic(fmt.Sprintf("unsupported address type %T", addr))
 	}
 }
+
+// SubjectAllowed is validating what types of resource can be used with the moved block.
+// At the time of writing, it was only ensuring that the moved blocks cannot be used against ephemeral resources.
+// This can later be expanded with more rules
+func (e *MoveEndpoint) SubjectAllowed() bool {
```

Comment:
This could definitely be more strict, afaik it should also not function on data blocks?

Reply:
It is allowed 🤔 
![Screenshot 2025-05-29 at 10 59 23](https://github.com/user-attachments/assets/7443d520-d51c-46da-9695-f89e25027549)

Am I wrong here?

Reply:
I think with ephemeral on the brain I got mixed up about what data is saved in a statefile

/facepalm